docs/lib/project-setup/fragments/android/rxjava/rxjava.md

```diff
@@ -0,0 +1,111 @@
+
+The Amplify library for Android and ships with APIs that return results via callbacks, as in:
+
+```java
+Post post = Post.builder()
+    .title("My First Post")
+    .build();
+
+Amplify.DataStore.save(post,
+    saved -> Log.i("MyAmplifyApp", "Saved a post."),
+    failure -> Log.e("MyAmplifyApp", "Save failed.", failure)
+);
+```
+
+Amplify also provides APIs that expose [Reactive Extensions](http://reactivex.io/), a cross-platform library for asynchronous and event-based programs:
+
+```java
+RxAmplify.DataStore.save(post)
+    .subscribe(
+        () -> Log.i("MyAmplifyApp", "Saved a post."),
+        failure -> Log.e("MyAmplifyApp", "Save failed.", failure)
+    );
+```
+
+While this doesn't save much for a single invocation, it provides great readability benefits when chaining asynchronous calls, since you can use standard RxJava operators to compose complex functionality into readable chunks. Consider a relational model where the creation of a `Post` requires also a creation of an `User` for the editor, and a `PostEditor` object to link the two together:
```

Comment:
```suggestion
While this doesn't save much for a single invocation, it provides great readability benefits when chaining asynchronous calls, since you can use standard RxJava operators to compose complex functionality into readable chunks. Consider a relational model where the creation of a `Post` requires also the creation of a `User` for the editor, and a `PostEditor` object to link the two together:
```